docs/direnv.md

```diff
@@ -0,0 +1,154 @@
+
+## What is Direnv and why use it?
+
+Direnv is an environment switcher for the shell. It knows how to hook into Bash, Zsh, and Fish shell to load or unload environment variables depending on the current directory. This allows project-specific environment settings to be loaded automatically as you move between different project directories.
+
+This is particularly useful in Haskell development as it enables automatic switching between different GHC versions and sets of dependencies, which is defined per project.
+
+## Prerequisites
+
+Ensure you have the following installed on your machine:
+
+- Nix
+- Direnv
+- Your choice of editor (VSCode, Emacs, or Vim)
+
+## Install and Configure Direnv
+
+Detailed instructions for installing Direnv can be found on the [official Direnv documentation](https://direnv.net/docs/installation.html). Depending on your operating system and preference, there may be package manager options, or you can install from source.
+
+Once Direnv is installed, you need to hook it into your shell. The process for this varies depending on what shell you use, but in general, you need to add a line to your shell initialization file (.bashrc, .zshrc, config.fish, etc.):
+
+```bash
+# Bash
+echo 'eval "$(direnv hook bash)"' >> ~/.bashrc
+
+# Zsh
+echo 'eval "$(direnv hook zsh)"' >> ~/.zshrc
+
+# Fish
+echo 'direnv hook fish | source' >> ~/.config/fish/config.fish
+```
+
+Remember to restart your shell or open a new terminal window to apply these changes.
+
+Next, you'll need to enable Direnv for your project directories. When you navigate into a directory with a `.envrc` file, Direnv will block the environment until you approve the changes by running `direnv allow`.
+
+### Pitfalls
+
+Direnv can be a bit tricky to get working. Some common issues include:
+
+- Forgetting to add the Direnv hook to your shell's configuration file.
+- Navigating to a directory before you've approved its `.envrc` file.
+- Forgetting to restart your shell or open a new terminal after setting up Direnv.
+
+Be sure to follow the installation and setup instructions carefully, and consult the [Direnv documentation](https://direnv.net/docs/troubleshooting.html) if you encounter problems.
+
+## Set Up Your Haskell Project
+
+Let's start by creating a simple HelloWorld Haskell application. First, navigate to your preferred working directory, then:
+
+```bash
+$ mkdir HelloWorld
+$ cd HelloWorld
+$ cabal init --is-executable
+```
+
+This will generate a basic `Main.hs` file with a HelloWorld program.
+
+## Configure Direnv for Your Project
+
+Create a `.envrc` file in your project root with the following content:
+
+```bash
+use flake "github:input-output-hk/devx#ghc8107-iog"
+```
+
```

Comment:
```suggestion
This will tell `direnv` to instantiate a shell with the same environment as `nix develop github:input-output-hk/devx#ghc8107-iog`, when changing into the folder where the `.envrc` resides.
```